script.metadata.editor/resources/lib/dialog_metadata.py

```diff
@@ -0,0 +1,322 @@
+#!/usr/bin/python
+# coding: utf-8
+
+########################
+
+from resources.lib.helper import *
+from resources.lib.json_map import *
+from resources.lib.functions import *
+
+########################
+
+class EditDialog(object):
+    def __init__(self,params):
+        self.params = params
+        self.dbid = params.get('dbid')
+        self.dbtype = params.get('type')
+
+        self.nfo_key = []
+        self.nfo_value = []
+
+        if self.dbtype in ['movie', 'tvshow', 'season', 'episode', 'musicvideo']:
+            library = 'Video'
+            self.nfo_support = True
+        else:
+            library = 'Audio'
+            self.nfo_support = False
+
+        self.method_details = '%sLibrary.Get%sDetails' % (library, self.dbtype)
+        self.param = '%sid' % self.dbtype
+        self.key_details = '%sdetails' % self.dbtype
+        self.properties = eval('%s_properties' % self.dbtype)
+
+        winprop('SelectDialogPreselect', clear=True)
+
+        self.init()
+
+    def init(self):
+        self.details = self.get_details()
+        self.file = self.details.get('file')
+
+        self.modeselect = []
+        self.keylist = []
+        self.presetlist = []
+        self.typelist = []
+        self.optionlist = []
+
+        self.generate_list()
+        self.editor_dialog()
```

Comment:
I just called it init() because after the editing of a value is done it returns to init() and fetches all made changes again to make them appear. It simply starts the addon from scratch without the __init__() step.